libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinGradleFinishBuildHandler.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.gradle.plugin
+
+import org.gradle.api.invocation.Gradle
+import org.gradle.api.logging.Logging
+import org.jetbrains.kotlin.compilerRunner.DELETED_SESSION_FILE_PREFIX
+import org.jetbrains.kotlin.compilerRunner.GradleCompilerRunner
+import org.jetbrains.kotlin.gradle.logging.kotlinDebug
+import org.jetbrains.kotlin.gradle.plugin.internal.state.TaskExecutionResults
+import org.jetbrains.kotlin.gradle.plugin.internal.state.TaskLoggers
+import org.jetbrains.kotlin.gradle.utils.relativeToRoot
+import org.jetbrains.kotlin.utils.addToStdlib.sumByLong
+import java.lang.management.ManagementFactory
+import kotlin.math.max
+
+// move to KotlinGradleBuildListener when min supported version is 6.1
+class KotlinGradleFinishBuildHandler {
+
+    private val log = Logging.getLogger(this.javaClass)
+    private var startMemory: Long? = null
+    private val shouldReportMemoryUsage = System.getProperty(KotlinGradleBuildServices.SHOULD_REPORT_MEMORY_USAGE_PROPERTY) != null
```

Comment:
Use `ObjectFactory.systemProperty` to get the value to allow reconfiguration if it changes.

Reply:
Should be `Providers.systemProperty(name)`